### PR TITLE
Replace separate_qautils_files with remove_non_production_files

### PR DESCRIPTION
**Pull request by marksliva**
The task was originally separating NON_PRODUCTION_FILES.txt out of the
bin_gpdb tarball and putting them into the QAUTILS_TARBALL, and leaving
OUTPUT_TARBALL with the rest of the files.
However, QAUTILS_TARBALL is never used, the only important function is
to remove the NON_PRODUCTION_FILES.txt.

With this commit, we refactor the separate_qautils_files to only remove
files. Also, we introduce a new pattern for TDD'ing bash scripts by
using the MIT-licensed BATS framework.

Co-authored-by: Mark Sliva <msliva@pivotal.io>
Co-authored-by: Amil Khanzada <akhanzada@pivotal.io>

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by marksliva** [comment]
gpdb pipeline:
https://releng.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-releng-refactor-non-prod-files-removal

gpdb open source pipeline:
https://releng.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb_opensource_release-refactor-non-prod-files-removal

File: concourse/scripts/remove_non_production_files.bats [hunk 1]
```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bats
```
Comment (baotingfang) [inline concourse/scripts/remove_non_production_files.bats]: 1. Do we need this testing file? 

2. Where did we install the bats testing framework?

For my opinion, We can remove this file.

File: concourse/scripts/remove_non_production_files.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+main() {
+  local tmp_dir="$(mktemp -d)"
+  local wd=$(pwd)
+
+  tar zxf "${INPUT_TARBALL}" -C "${tmp_dir}"
+
+  cd "${tmp_dir}"
+  xargs rm -f < "${wd}/${NON_PRODUCTION_FILES}"
+  if ! tar czf "${wd}/${OUTPUT_TARBALL}" * ; then
```
Comment (baotingfang) [inline concourse/scripts/remove_non_production_files.sh]: I suggest that we can verify the directory is empty or not in advance, and then execute the tar compression operation.

File: concourse/scripts/remove_non_production_files.bats (hunk 1 above)
Comment (amilkh) [reply concourse/scripts/remove_non_production_files.bats]: yes, we should check in the TDD tests to ensure future developers don't make the regression
please see https://github.com/greenplum-db/gpdb/pull/7992

File: concourse/scripts/remove_non_production_files.sh (hunk 2 above)
Comment (amilkh) [reply concourse/scripts/remove_non_production_files.sh]: good point! but, if $tmp_dir was empty, above `tar zxf "${INPUT_TARBALL}" -C "${tmp_dir}"` would have failed and we have `set -e`.

File: concourse/scripts/remove_non_production_files.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```
Comment (bradfordboyle) [inline concourse/scripts/remove_non_production_files.sh]: Small nitpick: can we rename the function to something like `remove_files` and pass the input tarball and list of files to remove as arguments to the function?

File: concourse/scripts/remove_non_production_files.bats (hunk 1 above)
Comment (bradfordboyle) [reply concourse/scripts/remove_non_production_files.bats]: I think this is a good example of using TDD w/ bash scripts. I vote that we keep it.

**Comment by marksliva** [comment]
cherry-picked to master: 967e223215744210cdea656655b75f5a9a6c4c8f

**Pull request merged**